spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/MessageGroupPreparedStatementSetter.java

```diff
@@ -18,26 +18,92 @@
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
+import java.sql.Types;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.core.serializer.support.SerializingConverter;
+import org.springframework.integration.IntegrationMessageHeaderAccessor;
 import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.integration.util.UUIDConverter;
 import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.jdbc.support.lob.DefaultLobHandler;
+import org.springframework.jdbc.support.lob.LobHandler;
 import org.springframework.messaging.Message;
 
 /**
  * Callback to be used with {@link JdbcChannelMessageStore}.
  * <p>
  * Behavior is same as standard {@link PreparedStatementSetter}, it takes in
- * extra {@code Message<?> requestMessage} and {@code Object groupId} as
- * parameter used for {@code addMessageToGroup} method in
- * {@link JdbcChannelMessageStore}.
+ * extra {@code Message<?> requestMessage}, {@code Object groupId}, {@code String region}
+ * and {@code boolean priorityEnabled} as
+ * parameters used for {@code addMessageToGroup} method in {@link JdbcChannelMessageStore}.
+ * Implementors of this class just can use the default constructor
+ * {@link #MessageGroupPreparedStatementSetter()} if default implementation of
+ * {@code SerializingConverter} and {@code LobHandler} is required.
  *
  * @author Meherzad Lahewala
  * @since 5.0
  * @see PreparedStatementSetter
  */
-@FunctionalInterface
-public interface MessageGroupPreparedStatementSetter {
+public class MessageGroupPreparedStatementSetter {
 
-	void setValues(PreparedStatement ps, Message<?> requestMessage, Object groupId) throws SQLException;
+	private static final Log logger = LogFactory.getLog(MessageGroupPreparedStatementSetter.class);
 
+	private final SerializingConverter serializer;
+
+	private final LobHandler lobHandler;
+
+	public MessageGroupPreparedStatementSetter(SerializingConverter serializer, LobHandler lobHandler) {
+		this.serializer = serializer;
+		this.lobHandler = lobHandler;
+	}
+
+	/**
+	 * Default Constructor using default serializer as
+	 * {@link SerializingConverter} and lobHandler as {@link DefaultLobHandler}.
+	 */
+	protected MessageGroupPreparedStatementSetter() {
+		this.serializer = new SerializingConverter();
+		this.lobHandler = new DefaultLobHandler();
```

Comment:
Thanks, @meherzad , for update.

Well, that's not what I want. I really would like to see them both as `null` here and add null checks to the `setValues()` method around them.
This way an implementor may decide to get some other logic for the message to store, e.g. convert to simple string representation. This way (s)he is able to override `setValues()` and don't count with serializer/lobHandler.

I think the way with nul checks we somehow minimize the time for the actions under `preparedStatement`.

Reply:
My only thinking to have serializer/lob handler as final to ensure the variables are not changed. But understand your point what of these variables are not required.

Reply:
Right, we just make like this:
```
protected MessageGroupPreparedStatementSetter() {
		this.serializer = null;
		this.lobHandler = null;
}
```
If you don't need them as `null`, you have another ctor for choice.